inference-engine/src/cldnn_engine/ops/reduce.cpp

```diff
@@ -89,7 +91,27 @@ void CreateReduceOp(Program& p, const std::shared_ptr<ngraph::Node>& op, cldnn::
         else if (rank - rawAxes.size() <= 4)
             out_format = cldnn::format::bfyx;
 
-        auto reorder_prim = cldnn::reorder(reorderLayerName, layerName, out_format, out_dt);
```

Comment:
Is it correct to apply your fix under `if (!keep_dims && rank > 4)` condition? Could the same problem occur with `rank == 4`?